classes/injector.php

```diff
@@ -61,6 +65,21 @@ public static function render_tracking_code(): string {
                 $type = $record->get_property('type');
                 if (key_exists($type, $plugins)) {
                     $tool = $plugins[$type]->get_tool_instance($record);
+                    if ($type === 'matomo') {
```

Comment:
This doesn't seem correct to me. You can create multiple instances of the matomo tool, so I think it should cycle through all of them and then if it can't find one then it auto creates outside the loop

Reply:
Also by design injector should know nothing about implementation details. Injector should just call an interface methods. 

I'd assume it should do something like 

```
if ($tool->support_auto_provision()) {
    $tool->auto_provision();
}
```

And then all details are encapsulated in a tool class.